internal/lang/functions.go

```diff
@@ -160,6 +160,12 @@ func (s *Scope) Functions() map[string]function.Function {
 			return s.funcs
 		})
 
+		s.funcs["templatestring"] = funcs.MakeTemplateStringFunc(s.BaseDir, func() map[string]function.Function {
+			// The templatestring function prevents recursive calls to itself
+			// by copying this map and overwriting the "templatestring" entry.
+			return s.funcs
```

Comment:
This logic isn't doing what the comment says. We should implement this.

Reply:
I think the recursive call for templatestring is being already handled (screenshots attached) so I need not to do anything explicitly for that.
![Uploading Screenshot 2024-02-08 at 1.51.54 PM.png…]()

Reply:
yeah sure will do that